src/ClientData/TimerDataTest.cpp

```diff
@@ -119,6 +119,22 @@ std::unique_ptr<TimerData> GetOrderedTimersSameDepth() {
   return timer_data;
 }
 
+std::unique_ptr<TimerData> GetUnorderedTimersSameDepth() {
+  std::unique_ptr<TimerData> timer_data = std::make_unique<TimerData>();
```

Comment:
IMO this is a great place to use auto, as the time is clear from the RHS of the expression.

Reply:
I think clang-tidy also suggests that

Reply:
TBH, it suggests it both ways. You have auto => use explicit type, you declare the type explicitly => use auto.